ArduPlane/config.h

```diff
@@ -78,6 +78,12 @@
  # error XXX
 #endif
 
+#ifndef FLIGHT_MODE_COUNT
```

Comment:
this should be a parameter to be generally useful I would think....but that means you have to hide the other unused flight mode parameters like an enable mode, I think...

Reply:
I agree with you but I'm not sure how to enable extra modes dynamically. I'd prefer it if this were a parameter, but this is much better than the default already, since it's a simple compile-time change of one number.

Maybe we can merge this and later on see how to make it dynamic?

Reply:
not my call on if it gets merged like this, but enabling params is done throughout the code...look at how RSSI params is enabled.......and it could be probably just either 6 modes or 9 modes with a simple enable param....6 is standard now with many TX have a 6P switch...its as much trouble to setup 9 ranges on the mode channel as 7....the first 6  fltmode params then and enable param for the following 3 additional params (note AP_PARAM_FLAG_ENABLE and the enable param group type...

Reply:
Hmm, you have a point, if you're enable extra modes, might as well enable 20 of them or whatnot, so it might as well be a single enable param. I'll wait for a bit more feedback to see how others want it too, thank you.

Reply:
less than 100us range is not going to be good IMHO....to easy to miss-setup or have noise in the TX and rX chain mess the mode up...

Reply:
Hmm, aren't protocols digital these days? I thought the channels were pretty accurate, but I can limit modes to 10 if that's the case.

Reply:
there are still people flying PPM and analog on ArduPilot....also, the mode switch may be a POT....having more than 10 positions to try to get to would be an issue I would think....maybe 12 would be acceptable using a 6P and a switch...I think this one will only be resolved in a DEVCALL...

Reply:
Ah, you're right about PPM. The way I use it is with OpenTX overrides on switches, which works great. A pot might be tricky.